base/loading.jl

```diff
@@ -3025,15 +3059,16 @@ end
                 @debug "Rejecting cache file $cachefile because it is for file $(includes[1].filename) not file $modpath"
                 return true # cache file was compiled from a different path
             end
-            for (modkey, req_modkey) in requires
-                # verify that `require(modkey, name(req_modkey))` ==> `req_modkey`
-                if identify_package(modkey, req_modkey.name) != req_modkey
-                    @debug "Rejecting cache file $cachefile because uuid mapping for $modkey => $req_modkey has changed"
-                    return true
-                end
-            end
+            # TODO Disabling this made `using Tables` only precompile once.
+            # for (modkey, req_modkey) in requires
+            #     # verify that `require(modkey, name(req_modkey))` ==> `req_modkey`
+            #     if identify_package(modkey, req_modkey.name) != req_modkey
+            #         @debug "Rejecting cache file $cachefile because uuid mapping for $modkey => $req_modkey has changed"
+            #         return true
+            #     end
+            # end
```

Comment:
Disabling this made the windows x86 tests pass.
But I haven't understood its impact yet ...

Reply:
Seems like it checking that the current environment gives the same dependency as the one in the cache file.

Reply:
For the `using Tables` example above `identify_package` gives back the correct pkg name `DataAPI`, but the uuid is wrong:
```sh
9a962f9c-6df0-11e9-0e5d-c546b8b5ee8a -- correct
9a965c9c-6df0-11e9-0e5d-c546b8b5ee8a -- wrong
```
The difference is only in the high part (`uuid >> 64`).

Reply:
I think the higher part of the uuid is a 32bit hash and the lower part is the 33bit buildid

Reply:
huh? Aren't these just the random uuids saved to the Project.tomls? https://github.com/JuliaData/DataAPI.jl/blob/b131356317cb3e5608d04a8bab7f98a1aa2701a7/Project.toml#L2

Reply:
Whee could that UUID come from? Some corruption from reading or some debug UUID created somewhere else etc?

Reply:
Just sanity checking binpack/binunpack, is this true on that platform?
```
u = UUID("9a962f9c-6df0-11e9-0e5d-c546b8b5ee8a")
UUID(UInt128(u)) == u
```

Reply:
Also worth checking this. I'm wondering if `normpath` on x86 windows isn't a noop in `replace_depot_path`
```
julia> b = Base.binpack(Base.PkgId(Base.UUID("9a962f9c-6df0-11e9-0e5d-c546b8b5ee8a"), "DataAPI"))
"\0\x8a\ued78F\xc5]\x0e\xe9\x11\xf0m\x9c/\x96\x9aDataAPI"

julia> Base.replace_depot_path(b) == b
true
```

Reply:
Indeed
```
julia> b = Base.binpack(pkg)
"\0\x8a\ued78F\xc5]\x0e\xe9\x11\xf0m\x9c/\x96\x9aDataAPI"

julia> Base.replace_depot_path(b)
"\0\x8a\ued78F\xc5]\x0e\xe9\x11\xf0m\x9c\\\x96\x9aDataAPI" # x9c/\x96 got replaced with x9c\\\x96
```
The `normpath` call inside `replace_depot_path` seems superfluous.

Will fix that tonight! Thanks all

Reply:
Hmm why was that only breaking x86 and not x64 windows..

Reply:
I believe it's because the Pkg tests will OOM on Windows x86 - @gbaraldi @staticfloat correct me if I'm wrong.

Reply:
yes that's correct.